src/amo/components/SponsoredAddonsShelf/index.js

```diff
@@ -29,17 +36,40 @@ type Props = {|
 
 export type InternalProps = {|
   ...Props,
+  _config: typeof config,
   _tracking: typeof tracking,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
   i18n: I18nType,
+  isLoading: boolean,
   resultsLoaded: boolean,
+  shelfData: SponsoredShelfType | null | void,
   shelves: { [shelfName: string]: Array<AddonType> | null },
 |};
 
 export class SponsoredAddonsShelfBase extends React.Component<InternalProps> {
   static defaultProps = {
+    _config: config,
     _tracking: tracking,
+    shelfData: undefined,
   };
 
+  componentDidMount() {
+    const { _config, isLoading, shelfData } = this.props;
+
+    if (
+      _config.get('enableFeatureUseAdzerkForSponsoredShelf') &&
```

Comment:
I suggest we add a method like `useAdzerk()` or something for this check. We'll be able to reuse it in `render()` and that might make the code slightly more readable.